### PR TITLE
Return Result in Engine::start

### PR DESCRIPTION
**Pull request by knkski**
Lets us handle startup errors in the engine more gracefully.

Signed-off-by: Kenneth Koski <knkski@bitwise.io>

### CODE REVIEW

**Comment by aludvik** [comment]
PRs should be opened against https://github.com/hyperledger/sawtooth-raft and https://github.com/hyperledger/sawtooth-pbft to make this change before this is merged.

**Comment by knkski** [comment]
@aludvik: https://github.com/hyperledger/sawtooth-pbft/pull/8 and https://github.com/hyperledger/sawtooth-raft/pull/42

File: sdk/examples/devmode_rust/src/engine.rs [hunk 1]
```diff
@@ -214,7 +214,7 @@ impl Engine for DevmodeEngine {
         updates: Receiver<Update>,
         service: Box<Service>,
         startup_state: StartupState,
-    ) {
+    ) -> Result<(), Error>{
```
Comment (ltseeley) [inline sdk/examples/devmode_rust/src/engine.rs]: Should probably be `Result<(), Error> {` instead of `Result<(), Error>{`

Comment (knkski) [reply sdk/examples/devmode_rust/src/engine.rs]: Good catch, updated 👍

**Pull request merged**